### PR TITLE
docs: Add Sublime Text LSP Guide

### LINKED ISSUE #521: Document SublimeText LSP Integration
Using [LSP for Sublime Text](https://lsp.sublimetext.io) we can use `esbonio`.

It is straightforward to configure. I will be proposing a pull request to document this.

### PR DESCRIPTION
**Pull request by vkhitrin**
Adding a guide to enable `esbonio` in Sublime Text
using LSP.

Resolve #521.

### CODE REVIEW

File: docs/lsp/editors/sublimetext-lsp/_installation.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Basic configuration for the language server is provided through the
+`LSP <https://github.com/sublimelsp/LSP>`_ plugin
+
+.. literalinclude:: ./editors/sublimetext-lsp/LSP.sublime-settings
+   :language: json
+   :start-at: set expandtab
+
+The server language executable can be overridden per project basis; refer to
+`Sublime Text Projects documentation <https://www.sublimetext.com/docs/projects.html>_`.
```
Comment (alcarney) [inline docs/lsp/editors/sublimetext-lsp/_installation.rst]:
```suggestion
`Sublime Text Projects documentation <https://www.sublimetext.com/docs/projects.html>`_.
```

File: docs/lsp/editors/sublimetext-lsp/_installation.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
+.. highlight:: none
+
+Basic configuration for the language server is provided through the
+`LSP <https://github.com/sublimelsp/LSP>`_ plugin
+
+.. literalinclude:: ./editors/sublimetext-lsp/LSP.sublime-settings
+   :language: json
+   :start-at: set expandtab
```
Comment (alcarney) [inline docs/lsp/editors/sublimetext-lsp/_installation.rst]:
```suggestion
   :start-at: {
```

File: docs/lsp/editors/sublimetext-lsp/_installation.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
+.. highlight:: none
+
+Basic configuration for the language server is provided through the
+`LSP <https://github.com/sublimelsp/LSP>`_ plugin
```
Comment (alcarney) [inline docs/lsp/editors/sublimetext-lsp/_installation.rst]:
```suggestion
`LSP <https://github.com/sublimelsp/LSP>`__ plugin
```

File: docs/lsp/getting-started.rst [hunk 4]
```diff
@@ -347,6 +363,9 @@ In the event that something does not work as expected, you can increase the logg
    Vim (vim-lsp)
       .. include:: ./editors/vim-lsp/_debugging.rst
 
+   Sublime Text (LSP)
+      .. include:: ./editors/sublimetext-lsp/_examples.rst
```
Comment (alcarney) [inline docs/lsp/getting-started.rst]:
```suggestion
      .. include:: ./editors/sublimetext-lsp/_debugging.rst
```

File: docs/lsp/editors/sublimetext-lsp/_debugging.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+Refer to the troubleshooting guide of `LSP <https://lsp.sublimetext.io/troubleshooting/>`_.
```
Comment (alcarney) [inline docs/lsp/editors/sublimetext-lsp/_debugging.rst]:
```suggestion
Refer to the troubleshooting guide of `LSP <https://lsp.sublimetext.io/troubleshooting/>`__.
```

File: docs/lsp/editors/sublimetext-lsp/_examples.rst [hunk 6]
```diff
@@ -0,0 +1,8 @@
+To try the example configuration on your machine.
+
+#. Ensure you have `Package Control <https://packagecontrol.io>`_
+   installed.
+#. Install `LSP <https://packagecontrol.io/packages/LSP>`_ package.
```
Comment (alcarney) [inline docs/lsp/editors/sublimetext-lsp/_examples.rst]:
```suggestion
#. Install `LSP <https://packagecontrol.io/packages/LSP>`__ package.
```

**Review by alcarney — COMMENTED** [review]
Awesome, thanks for this! :smile: 

Made some suggestions, mostly fixing typos or build warnings but other than that, it looks good to me

**Comment by vkhitrin** [comment]
@alcarney,
Thanks for the quick review and catching the typos!
They all should be resolved now.

File: docs/lsp/editors/sublimetext-lsp/_examples.rst (hunk 6 above)
Comment (vkhitrin) [reply docs/lsp/editors/sublimetext-lsp/_examples.rst]: Resolved.

File: docs/lsp/editors/sublimetext-lsp/_debugging.rst (hunk 5 above)
Comment (vkhitrin) [reply docs/lsp/editors/sublimetext-lsp/_debugging.rst]: Resolved.

File: docs/lsp/getting-started.rst (hunk 4 above)
Comment (vkhitrin) [reply docs/lsp/getting-started.rst]: Resolved.

File: docs/lsp/editors/sublimetext-lsp/_installation.rst (hunk 3 above)
Comment (vkhitrin) [reply docs/lsp/editors/sublimetext-lsp/_installation.rst]: Resolved.

File: docs/lsp/editors/sublimetext-lsp/_installation.rst (hunk 2 above)
Comment (vkhitrin) [reply docs/lsp/editors/sublimetext-lsp/_installation.rst]: Resolved.

File: docs/lsp/editors/sublimetext-lsp/_installation.rst (hunk 1 above)
Comment (vkhitrin) [reply docs/lsp/editors/sublimetext-lsp/_installation.rst]: Resolved.

**Comment by alcarney** [comment]
No worries, thanks again for taking the time to write this up :slightly_smiling_face:

**Pull request merged**